modules/engine/src/listeners.ts

```diff
@@ -542,6 +546,55 @@ export async function resolveExistingWithdrawals(
   );
 }
 
+export async function submitUnsubmittedWithdrawals(
+  channel: FullChannelState,
+  store: IEngineStore,
+  chainAddresses: ChainAddresses,
+  chainService: IVectorChainService,
+  logger: BaseLogger,
+): Promise<void> {
+  const method = "submitUnsubmittedWithdrawals";
+  const methodId = getRandomBytes32();
+
+  const withdrawInfo = await chainService.getRegisteredTransferByName(
+    TransferNames.Withdraw,
+    chainAddresses[channel.networkContext.chainId].transferRegistryAddress,
+    channel.networkContext.chainId,
+  );
+  if (withdrawInfo.isError) {
+    logger.error(
+      { method, methodId, error: withdrawInfo.getError()?.toJson() },
+      "Error in chainService.getRegisteredTransferByName",
+    );
+  }
+
+  const unsubmitted = await store.getUnsubmittedWithdrawals(channel.channelAddress, withdrawInfo.getValue().definition);
+  for (const u of unsubmitted) {
+    logger.info({ method, methodId, transferId: u.transfer.transferId }, "Submitting unsubmitted withdrawal");
+    try {
+      const commitment = await WithdrawCommitment.fromJson(u.commitment);
+      const tx = await chainService.sendWithdrawTx(channel, commitment.getSignedTransaction());
+      if (tx.isError) {
+        logger.error(
+          { method, methodId, error: tx.getError()?.toJson(), commitment: u },
+          "Error in chainService.sendWithdrawTx",
+        );
+      }
+      logger.info(
+        { method, methodId, transactionHash: tx.getValue().transactionHash },
+        "Submitted unsubmitted withdrawal",
+      );
+      commitment!.addTransaction(tx.getValue().transactionHash);
+      await store.saveWithdrawalCommitment(u.transfer.transferId, commitment!.toJson());
```

Comment:
will the counterparty get notice of the changed withdrawal hash? maybe that's not so important, but would be nice

Reply:
Potentially: don't notify tx hash at all until tx complete?

Notify on submit, don't include hash anymore, since unconfirmed

Reply:
> Potentially: don't notify tx hash at all until tx complete?
It won't actually, right? Because the tx service method won't return until confirmed.